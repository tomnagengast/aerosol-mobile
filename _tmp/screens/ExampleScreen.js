/* eslint-disable no-unused-vars, class-methods-use-this, indent, import/extensions */

import React from 'react';
import { Text, View } from 'react-native';

const styles = {
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
    placeholder: {
        fontSize: 36,
    },
    courseList: {},
    courseName: {
        fontWeight: '800',
        fontSize: 24,
    },
};

export default class ExampleScreen extends React.Component {
    state = {
        loaded: false,
        courses: {},
    };

    async componentWillMount() {
        try {
            const response = await axios.get('courses');
            this.setState({ courses: response.data.courses, loaded: true });
        } catch (error) {
            console.log({ error });
        }
    }

    render() {
        return !this.state.loaded
            ? (
                <View style={styles.container}>
                    <Text style={styles.placeholder}>Hang Tight! We're grabbing our clubs</Text>
                </View>
            ) : (
                <View style={styles.container}>
                    <Text>Welcome Screen</Text>
                    <Text>Environment: {process.env.NODE_ENV}</Text>
                    <View style={styles.courseList}>
                        {this.state.courses.map(course => (
                            <Text key={course.id} style={styles.courseName}>{course.name}</Text>
                        ))}
                    </View>
                </View>
            );
    }
}
