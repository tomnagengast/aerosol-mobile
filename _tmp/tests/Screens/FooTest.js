/* eslint-disable */
import { View, Text } from 'react-native';
// jest.mock('react-native-text', () => 'Text');
import ExampleScreen from '../../_tmp/ExampleScreen';

// it('should render three <Foo /> components', () => {
//     const wrapper = shallow(<ExampleScreen />);
//     expect(wrapper.find(ExampleScreen)).debug().toHaveLength(0);
// });
//
describe('<ExampleScreen />', () => {
    it('should render a label', () => {
        const wrapper = shallow(<Text>Hang Tight! We're grabbing our clubs...</Text>);
        expect(wrapper).toMatchSnapshot();
    });


    //        <View className="container">
    //            <Text className="placeholder">Hang Tight! We're grabbing our clubs...</Text>
    //        </View>
    //    )).to.equal(true);

    //     it('should render without throwing an error', () => {
    //         // assert(1 === 1, '1 is 1');
    //         expect(shallow(<Foo/>).contains(<div className="foo">Bar</div>)).toBe(true);
    //     });
    //
    //     it('renders using Snapshots', () => {
    //         expect(renderer.create(
    //             <div className="foo">Bar</div>
    //         )).toMatchSnapshot();
    //     });
    //
    //     mount: good for checking prop values, maybe state?
    //     it('allows us to set props', () => {
    //         const wrapper = mount(<Foo bar="baz"/>);
    //         expect(wrapper.props().bar).to.equal('baz');
    //         wrapper.setProps({ bar: 'foo' });
    //         expect(wrapper.props().bar).to.equal('foo');
    //     });
    //
    //     it('simulates click events', () => {
    //         const onButtonClick = sinon.spy();
    //         const wrapper = mount((
    //             <Foo onButtonClick={onButtonClick}/>
    //         ));
    //         wrapper.find('button').simulate('click');
    //         expect(onButtonClick).to.have.property('callCount', 1);
    //     });
});

