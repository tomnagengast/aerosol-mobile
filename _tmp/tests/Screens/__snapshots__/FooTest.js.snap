// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleScreen /> should render a label 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Hang Tight! We're grabbing our clubs...
</Text>
`;

exports[`<Foo /> should render a label 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Hang Tight! We're grabbing our clubs...
</Text>
`;

exports[`should render a label 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Hang Tight! We're grabbing our clubs...
</Text>
`;
